core/src/main/java/org/elasticsearch/common/transport/TransportAddress.java

```diff
@@ -46,5 +46,9 @@
 
     boolean sameHost(TransportAddress other);
 
+    default boolean isLocalAddress() {
```

Comment:
I do not see the issue with the default implementation? I did rename the method though in 6f138460172805fb0665c5ccba85846151992be4.